src/screens/SUE/SueReportScreen.tsx

```diff
@@ -41,17 +41,27 @@ type TRequiredFields = {
 };
 
 const sueProgressWithRequiredInputs = (
-  progress: TProgress[],
-  fields: TRequiredFields
+  fields: TRequiredFields,
+  geoMap: { locationIsRequired: boolean; locationStreetIsRequired: boolean },
+  progress: TProgress[]
 ): TProgress[] => {
   const requiredInputs: { [key: string]: boolean } = {};
 
-  for (const section in fields) {
-    for (const field in fields[section]) {
-      requiredInputs[field] = fields[section][field];
+  if (fields?.contact) {
+    for (const field in fields.contact) {
+      requiredInputs[field] = fields.contact[field];
     }
   }
 
+  if (geoMap?.locationIsRequired) {
+    requiredInputs['city'] = true;
+    requiredInputs['postalCode'] = true;
+  }
+
+  if (geoMap?.locationStreetIsRequired) {
+    requiredInputs['street'] = true;
+  }
```

Comment:
what about

```
requiredInputs['city'] = !!geoMap?.locationIsRequired;
requiredInputs['postalCode'] = !!geoMap?.locationIsRequired;
requiredInputs['street'] = !!geoMap?.locationStreetIsRequired;
```

Reply:
In requiredInputs add `city`, `postalCode` and `street` and set their values to `true`. Otherwise this data will not be in the object. This also works with the `requiredInputs.city` logic. Now I thought, we use these texts in different parts of the code. Maybe we should create `const` to update these texts in one place.

```
SUE.INPUTS.CITY = 'city'
SUE.INPUTS.POSTALCODE = 'postalCode'
SUE.INPUTS.STREET = 'street'
```

Reply:
i don't understand how this relates to my suggestion of refactoring?

Reply:
Sure, this can be an additional improvement to avoid the "hardcoded" strings in multiple places.
(POSTAL_CODE not POSTALCODE)

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1147

Reply:
> requiredInputs['city'] = !!geoMap?.locationIsRequired;
> requiredInputs['postalCode'] = !!geoMap?.locationIsRequired;
> requiredInputs['street'] = !!geoMap?.locationStreetIsRequired;

That's exactly the right point of view. 👍

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/80a3dd450ed7e72972b8a37f2dc61734b19cea64